modules/video_writer.py

```diff
@@ -0,0 +1,76 @@
+import cv2
+import numpy as np
+
+
+class VideoWriter:
+
+    def __init__(self, outfile=None):
+        self.outfile = outfile
+        self.frame_resolution = None
+        self.fps = 30  # TODO: read from video file eventually
+        self.writer = None
+
+    def write(self, frame_2d, frame_3d):
+        if not self.outfile:
+            return
+
+        # lazy-init when first frame is written
+        if not self.writer:
+            self.init_writer(frame_2d)
+
+        frame = self.combine(frame_2d, frame_3d)
+        self.writer.write(frame)
+
+    def init_writer(self, frame_2d):
+        print(f"Output will be written to {self.outfile}")
+        fourcc = cv2.VideoWriter_fourcc(*'mp4v')  # appears to work fine when outfile is *.mp4 or *.mov
+        height, width, _ = frame_2d.shape
+
+        self.frame_resolution = height, width  # numpy notation
+        resolution_combined = (width * 2, height)  # opencv notation
+
+        self.writer = cv2.VideoWriter(self.outfile, fourcc, self.fps, resolution_combined)
+
+    def release(self):
+        if self.outfile and self.writer:
+            self.writer.release()
+            self.writer = None
+            print(f"Finished writing out file to {self.outfile}")
+
+    def combine(self, frame_2d, frame_3d):
+        height, width = self.frame_resolution
+
+        frame_3d = self.pad(frame_3d, height, width)
+        frame_3d = self.crop(frame_3d, height, width)
+        return  np.concatenate((frame_2d, frame_3d), axis=1)
+
+    @staticmethod
+    def pad(frame, height, width):
```

Comment:
I'm not sure wether it should be static. It definetely can, but it looks like implementation detail, so for me it is a private one (and `write` is the only one public method).